### PR TITLE
feat: check conflicts on postgres

### PR DESCRIPTION
**Pull request by carneiro-cw**

### CODE REVIEW

File: src/eth/storage/postgres/postgres.rs [hunk 1]
```diff
@@ -471,23 +484,35 @@ impl EthStorage for Postgres {
...
-                        block_number
+                        block_number,
+                        original_nonce,
+                        original_balance
                     )
                     .execute(&mut *tx)
                     .await
                     .context("failed to insert account")?;
 
+                    if account_result.rows_affected() != 1 {
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:506]: can you create a function to represent this case? or at least add some comments

File: src/eth/primitives/execution_conflict.rs [hunk 2]
```diff
@@ -58,4 +67,16 @@ pub enum ExecutionConflict {
...
+    Account {
+        address: Address,
+        expected_nonce: BigDecimal,
+        expected_balance: BigDecimal,
+    },
+
+    PgSlot {
+        address: Address,
+        slot: Vec<u8>,
+        expected: Vec<u8>,
```
Comment (arthur-cw) [inline src/eth/primitives/execution_conflict.rs:80]: why not use [u8; 32] if it's gonna be converted later?

File: src/eth/storage/postgres/postgres.rs [hunk 3]
```diff
@@ -508,25 +533,35 @@ impl EthStorage for Postgres {
...
                             change.address.as_ref(),
-                            block_number
+                            block_number,
+                            original_value
                         )
                         .execute(&mut *tx)
                         .await
                         .context("failed to insert slot")?;
 
+                        if slot_result.rows_affected() != 1 {
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:555]: same as above

File: src/eth/storage/postgres/postgres.rs (hunk 1 above)
Comment (carneiro-cw) [reply src/eth/storage/postgres/postgres.rs:506]: Sure

File: src/eth/primitives/execution_conflict.rs (hunk 2 above)
Comment (carneiro-cw) [reply src/eth/primitives/execution_conflict.rs:80]: True

**Review by arthur-cw — APPROVED** [review]

**Pull request merged**